OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs

```diff
@@ -32,6 +34,18 @@ public class DeployToUpgradeInfo : ITraitInfo, Requires<UpgradeManagerInfo>
 		[Desc("Cursor to display when unable to (un)deploy the actor.")]
 		public readonly string DeployBlockedCursor = "deploy-blocked";
 
+		[Desc("Facing that the actor must face before deploying. Will deploy disregarding facing by default.")]
+		public readonly int? Facing = null;
```

Comment:
It was, but @Phrohdoh made a somewhat good case for using `int?` instead, and thus not needing to use (the otherwise valid value) -1 as a magic invalid default.